lib/Driver/Driver.cpp

```diff
@@ -342,6 +344,13 @@ static bool populateOutOfDateMap(InputInfoMap &map, StringRef argsHashStr,
   }
 
   if (!optionsMatch) {
+    if (ShowIncrementalBuildDecisions) {
+      llvm::outs() << "Incremental compilation has been disabled, because "
+                   << "different arguments were passed to the compiler.\n"
+                   << "\tCompiling with argument hash: " << argsHashStr << "\n"
```

Comment:
I'm not sure printing the hash is useful. There's not really anything people can do with it.